### PR TITLE
ESQL: Entirely drop `version` field

### PR DESCRIPTION
**Pull request by nik9000**
This field is no longer supported.

### CODE REVIEW

**Comment by nik9000** [comment]
Note that this is targeting `8.14` and not `main`. We'll forward port this once serverless is ready.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/10_basic.yml [hunk 1]
```diff
@@ -333,3 +333,15 @@ setup:
...
+version is not allowed:
+  - requires:
+      cluster_features: ["gte_v8.14.0"]
+      reason: version allowed in 8.13.latest
+  - do:
+      catch: /unknown field \[version\]/
+      esql.query:
+        body:
+          query: 'from test'
+          version: cat
```
Comment (alex-spies) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/10_basic.yml:347]: I wanted to say that we could also just use the `EsqlQueryRequestTests`, but this is better.

**Review by alex-spies — APPROVED** [review]
Thanks, Nik!

I think we can also remove the respective tests in `EsqlQueryRequestTests`:
- testAnyVersionIsValid
- testMissingVersionIsValid

**Comment by nik9000** [comment]
> I think we can also remove the respective tests in `EsqlQueryRequestTests`:

:+1:

**Pull request merged**